contracts/src/v0.1/L2endpoint.sol

```diff
@@ -92,44 +76,6 @@ contract L2Endpoint is Ownable {
         return (requestId, preSeed);
     }
 
-    /**
-     * @dev calls target address with exactly gasAmount gas and data as calldata
-     * or reverts if at least gasAmount gas is not available.
-     */
-    function callWithExactGas(
-        uint256 gasAmount,
-        address target,
-        bytes memory data
-    ) internal returns (bool success) {
-        // solhint-disable-next-line no-inline-assembly
-        assembly {
-            let g := gas()
-            // Compute g -= GAS_FOR_CALL_EXACT_CHECK and check for underflow
-            // The gas actually passed to the callee is min(gasAmount, 63//64*gas available).
-            // We want to ensure that we revert if gasAmount >  63//64*gas available
-            // as we do not want to provide them with less, however that check itself costs
-            // gas.  GAS_FOR_CALL_EXACT_CHECK ensures we have at least enough gas to be able
-            // to revert if gasAmount >  63//64*gas available.
-            if lt(g, GAS_FOR_CALL_EXACT_CHECK) {
-                revert(0, 0)
-            }
-            g := sub(g, GAS_FOR_CALL_EXACT_CHECK)
-            // if g - g//64 <= gasAmount, revert
-            // (we subtract g//64 because of EIP-150)
-            if iszero(gt(sub(g, div(g, 64)), gasAmount)) {
-                revert(0, 0)
-            }
-            // solidity calls check that a contract actually exists at the destination, so we do the same
-            if iszero(extcodesize(target)) {
-                revert(0, 0)
-            }
-            // call and return whether we succeeded. ignore return data
-            // call(gas,addr,value,argsOffset,argsLength,retOffset,retLength)
-            success := call(gasAmount, target, 0, add(data, 0x20), mload(data), 0, 0)
-        }
-        return success;
-    }
-
     function requestRandomWords(
```

Comment:
@KelvinThai You are planning to move this and other VRF related functions to the `contracts/src/v0.1/L2EndpointBase.sol` in the followup PR, right?

Reply:
yes, after all core functions is done, I'm gonna follow up with a code cleaning PR